WordPress/wordcamp.org
# Tools: Try using `phpcs-changed` to limit php linting to changed lines

We have a lot of large files with a lot of errors, and while we can go through and commit fixes to these while working, it's not feasible to fix all errors this way. This introduces [phpcs-changed](https://github.com/sirbrillig/phpcs-changed) to only report on changed lines in a PR. By changing travis to only sniff changed lines, we could actually require that CI passes before merging ✅

By default, `phpcs-changed --git` works on staged changes (so you could use it locally pre-commit, I think); but it needs to be called manually to diff between branch changes. That's where `phpcs-branch.php` comes in, travis calls that script, which uses the diff between `TRAVIS_BRANCH` (the PR target) and `HEAD` (the branch) to get the changes in each modified file. It also runs phpcs directly for added files (a better dev-ex, otherwise the file is null).

### How to test the changes in this Pull Request:

You can run the phpcs-branch file directly, if you want, but you need to define the target branch: `export TRAVIS_BRANCH='production'; php .github/bin/phpcs-branch.php`

Otherwise the CI check will be the test. This should report the newly introduced errors (will fix those commits before merging, of course)
